cmd/insertvariantobject/insertvariantobject.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"encoding/json"
+
+	sf "github.com/snowflakedb/gosnowflake"
+)
+
+func main() {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
+
+	cfg, err := sf.GetConfigFromEnv([]*sf.ConfigParam{
+		{Name: "Account", EnvName: "SNOWFLAKE_TEST_ACCOUNT", FailOnMissing: true},
+		{Name: "User", EnvName: "SNOWFLAKE_TEST_USER", FailOnMissing: true},
+		{Name: "Password", EnvName: "SNOWFLAKE_TEST_PASSWORD", FailOnMissing: true},
+		{Name: "Warehouse", EnvName: "SNOWFLAKE_TEST_WAREHOUSE", FailOnMissing: true},
+		{Name: "Database", EnvName: "SNOWFLAKE_TEST_DATABASE", FailOnMissing: true},
+		{Name: "Schema", EnvName: "SNOWFLAKE_TEST_SCHEMA", FailOnMissing: true},
+		{Name: "Host", EnvName: "SNOWFLAKE_TEST_HOST", FailOnMissing: false},
+		{Name: "Port", EnvName: "SNOWFLAKE_TEST_PORT", FailOnMissing: false},
+		{Name: "Protocol", EnvName: "SNOWFLAKE_TEST_PROTOCOL", FailOnMissing: false},
+	})
+	if err != nil {
+		log.Fatalf("failed to create Config, err: %v", err)
+	}
+	dsn, err := sf.DSN(cfg)
+	if err != nil {
+		log.Fatalf("failed to create DSN from Config: %v, err: %v", cfg, err)
+	}
+
+	db, err := sql.Open("snowflake", dsn)
+	if err != nil {
+		log.Fatalf("failed to connect. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+
+	param := map[string]string{"key": "value"}
+	jsonStr, _ := json.Marshal(param)
+
+	createTableQuery := "CREATE OR REPLACE TABLE insertvariantobject (c1 VARIANT, c2 OBJECT);"
+
+	// https://docs.snowflake.com/en/sql-reference/functions/parse_json
+	// can do with TO_VARIANT(PARSE_JSON(..)) as well, but PARSE_JSON already produces VARIANT
+	insertQuery := "INSERT INTO insertvariantobject (c1, c2) SELECT PARSE_JSON(?), TO_OBJECT(PARSE_JSON(?));"
+	// https://docs.snowflake.com/en/sql-reference/data-types-semistructured#object
+	insertOnlyObject := "INSERT INTO insertvariantobject (c2) SELECT OBJECT_CONSTRUCT('name', 'Jones'::VARIANT,'age',  42::VARIANT);"
+
+	selectQuery := "SELECT c1, c2 FROM insertvariantobject;"
+
+	fmt.Printf("Running CREATE OR REPLACE TABLE\n")
+	db.Exec(createTableQuery)
```

Comment:
If you call many operations on `db` directly, it causes authentication every time. It would be better, if we had connection created here.

Reply:
again this is a very useful pointer, thank you !
however the example is only here because i thought it would be better as a code, instead of the documentation. by no means is it representative of any production-ready approach, it is really intended to show how you can insert VARIANT and OBJECT into your table with variable binding.

Reply:
On the other hand, I like keeping code clean everywhere it is possible. Otherwise, someone may copy/use the code incorrectly, just by looking at the example.

Reply:
has been rewritten eventually to use `db.Conn`